### PR TITLE
Rename abstract test classes

### PR DESCRIPTION
**Pull request by tresat**
Adjust the naming to make it clear which test fixtures are abstract classes meant for extension.

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

File: subprojects/architecture-test/src/changes/archunit_store/public-api-mutable-properties.txt [hunk 1]
```diff
@@ -901,4 +901,4 @@ Method <org.gradle.vcs.VersionControlRepository.getRootDir()> does not have raw
 Method <org.gradle.vcs.VersionControlSpec.getRepoName()> does not have raw return type assignable to org.gradle.api.provider.Provider in (VersionControlSpec.java:0)
 Method <org.gradle.vcs.VersionControlSpec.getRootDir()> does not have raw return type assignable to org.gradle.api.provider.Property in (VersionControlSpec.java:0)
 Method <org.gradle.vcs.VersionControlSpec.getUniqueId()> does not have raw return type assignable to org.gradle.api.provider.Provider in (VersionControlSpec.java:0)
-Method <org.gradle.vcs.git.GitVersionControlSpec.getUrl()> does not have raw return type assignable to org.gradle.api.provider.Property in (GitVersionControlSpec.java:0)
\ No newline at end of file
+Method <org.gradle.vcs.git.GitVersionControlSpec.getUrl()> does not have raw return type assignable to org.gradle.api.provider.Property in (GitVersionControlSpec.java:0)
```
Comment (wolfs) [inline subprojects/architecture-test/src/changes/archunit_store/public-api-mutable-properties.txt:904]: ❔ What is the change here? A line ending difference?

**Review by wolfs — APPROVED** [review]

File: subprojects/architecture-test/src/changes/archunit_store/public-api-mutable-properties.txt (hunk 1 above)
Comment (tresat) [reply subprojects/architecture-test/src/changes/archunit_store/public-api-mutable-properties.txt:904]: Yes, that's my precommit hook ensuring a newline is present at the end of a file upon change.  The change itself was accidentally added when cherrypicking commits and later reverted.

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**